pkg/gardenadm/cmd/bootstrap/bootstrap.go

```diff
@@ -46,7 +52,44 @@ gardenadm bootstrap --kubeconfig ~/.kube/config`,
 	return cmd
 }
 
-func run(_ context.Context, opts *Options) error {
-	opts.Log.Info("Not implemented as well")
+// NewClientSetFromFile in alias for botanist.NewClientSetFromFile.
+// Exposed for unit testing.
+var NewClientSetFromFile = botanist.NewClientSetFromFile
+
+func run(ctx context.Context, opts *Options) error {
+	clientSet, err := NewClientSetFromFile(opts.Kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed creating client: %w", err)
+	}
+
+	if err := ensureNoGardenletOrOperator(ctx, clientSet.Client()); err != nil {
+		return err
+	}
+
+	opts.Log.Info("Command is work in progress")
+	return nil
+}
+
+// ensureNoGardenletOrOperator is a safety check that prevents operators from accidentally executing
+// `gardenadm bootstrap` on a cluster that is already used as a runtime cluster with gardener-operator or as a seed
+// cluster. Doing so would lead to conflicts when `gardenadm bootstrap` starts deploying components like provider
+// extensions.
+func ensureNoGardenletOrOperator(ctx context.Context, c client.Reader) error {
+	for _, key := range []client.ObjectKey{
+		{Namespace: v1beta1constants.GardenNamespace, Name: "gardener-operator"},
+		{Namespace: v1beta1constants.GardenNamespace, Name: "gardenlet"},
+	} {
+		if err := c.Get(ctx, key, &appsv1.Deployment{}); err != nil {
```

Comment:
Looking at it, we could alternatively also use https://github.com/gardener/gardener/blob/1a53fc3b9367c90126c9aa17e3f9fcfe5c1d2a4f/pkg/utils/kubernetes/object.go#L64 to check if there is any `Deployment` with `app=gardener` label. WDYT?

Reply:
That check might be too broad if we deploy other gardener components with `gardenadm bootstrap` later. 
I could use the function with a more specific label selector. But then, the value over specific names might be little.
WDYT?

Reply:
If you refer to GRM - this friend has `app=gardener-resource-manager` (for whatever reason). However, I agree that it could be problematic if we decide to change its label. Then let's keep it as-is for now.